javascript/first-js-oop-project.md

```diff
@@ -0,0 +1,142 @@
+description: "Victorious war tour."},
+{name: "Jenny Lewis",
+description: "On the line tour 2019."},
+]
+```
+- You can also try creating an event class and than enter the details.
+- You can try the belo code for the same.
+```javascript
+  class Event {
+  constructor(name, description) {
+    this.name = name;
+    this.description = description;
+  }
+  let Event = new Event("KLOS Golden Gala", "An evening with hollywood vampires");
+```
+
+- keep checking everytime you push the element using ***console.log***.
+```javascript
+  //inorder to check whether the elements are pushed, Use Console.log
+ console.log(EventArray);
+```
+
+- After that you have created the array now we need to write the jQuery code to iterate through it in the same js file event.js .
+- The jQuery library is a single JavaScript file, and you reference it with the HTML.
+```html
+<head>
+<script src="jquery-3.4.0.min.js"></script>
+</head>
+```
+
+- If an object is used as the collection, the callback is passed a key-value pair each time:
+```javascript
+ var obj = {
+  "KLOS Golden Gala": "An evening with hollywood vampires",
+  "Jenny Lewis": "On the line tour 2019"
+};
+$.each( obj, function( key, value ) {
+  alert( key + ": " + value );
+});
+```
+- Below is the jQuery code to iterate through the array events that you are asked to create.
+- ***.each()*** is used to iterate through the array of objects. Above code represents the syntax to iterate using ***.each()***
+- ***.html()*** is used to return the HTML code from jQuery to the target element of the HTML page.
+  - "event" be the target element in the below code.
+
+  ```javascript
+   $(document).ready(function(){
+    var html = "";
+    $.each(events, function(itemName, item){
+        $.each(item, function(subItemName, subItem){
+             html += ("<h1>"+ subItem + "</h1>"); });
+        });
+        // insert final html into #events...
+    $("#event").html(html); });```
+
+- You might have noticed that all jQuery methods in our examples, are inside a document ready event.
+- This is to prevent any jQuery code from running before the document is finished loading(is ready).
+
+6. Create an HTML file named ***index.html***. 
+- This is the HTML file containing the div tag or that can be any other element as per the requirement.
+
+```html
+<!DOCTYPE html>
+<html>
+    <head>
+        <title></title>
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+    </head>
+    <body>
+            <div id="events">
+                    <!-- EVENTS GET INSERTED HERE -->
+            </div>
+    </body>
+    <script src = "./event.js"></script>
+</html>
+```
+
+
+
+### Independent Practice
+
+- Make changes in the structure of array like adding or removing elements or objects, to understand the iteration process.
+- Go through the concept of [each](http://api.jquery.com/jquery.each/) in jQuery which will allow you to know more about array.
```

Comment:
Use <code>\`.each()`</code> to show that it's a part of code.

Reply:
☝🏻